azure/services/scalesets/scalesets.go

```diff
@@ -363,6 +363,28 @@ func (s *Service) validateSpec(ctx context.Context) error {
 		}
 	}
 
+	// Validate DiagnosticProfile spec
+	if spec.DiagnosticsProfile.Boot != nil {
+		if spec.DiagnosticsProfile.Boot.StorageAccountType == infrav1.UserManagedDiagnosticsStorage {
+			if spec.DiagnosticsProfile.Boot.UserManaged == nil {
+				return azure.WithTerminalError(fmt.Errorf("userManaged must be specified when storageAccountType is '%s'", infrav1.UserManagedDiagnosticsStorage))
+			} else if spec.DiagnosticsProfile.Boot.UserManaged.StorageAccountURI == "" {
+				return azure.WithTerminalError(fmt.Errorf("storageURI cannot be empty when storageAccountType is '%s'", infrav1.UserManagedDiagnosticsStorage))
+			}
+		}
+
+		possibleStorageAccountTypeValues := []string{
+			string(infrav1.DisabledDiagnosticsStorage),
+			string(infrav1.ManagedDiagnosticsStorage),
+			string(infrav1.UserManagedDiagnosticsStorage),
+		}
+
+		if !slice.Contains(possibleStorageAccountTypeValues, string(spec.DiagnosticsProfile.Boot.StorageAccountType)) {
+			return azure.WithTerminalError(fmt.Errorf("invalid storageAccountType: %s. Allowed values are %v",
+				spec.DiagnosticsProfile.Boot.StorageAccountType, possibleStorageAccountTypeValues))
+		}
```

Comment:
This validation should be done by the openapi, do we need to duplicate it here?

Reply:
Perhaps. I noticed it's present for some recent features and not for others.
I'd be happy to remove it if not necessary but I put there to get a discussion going over it with maintainers.